src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -449,61 +398,102 @@ public string OutputAssembly
                 }
             }
         }
-        internal string outputAssembly = null;
+        private string outputAssembly = null;
 
         /// <summary>
         /// The output type of the assembly.
         /// </summary>
-        [Parameter(ParameterSetName = "FromSource")]
-        [Parameter(ParameterSetName = "FromMember")]
-        [Parameter(ParameterSetName = "FromPath")]
-        [Parameter(ParameterSetName = "FromLiteralPath")]
+        [Parameter(ParameterSetName = FromSourceParameterSetName)]
+        [Parameter(ParameterSetName = FromMemberParameterSetName)]
+        [Parameter(ParameterSetName = FromPathParameterSetName)]
+        [Parameter(ParameterSetName = FromLiteralPathParameterSetName)]
         [Alias("OT")]
-        public OutputAssemblyType OutputType
-        {
-            get
-            {
-                return outputType;
-            }
-            set
-            {
-                outputTypeSpecified = true;
-                outputType = value;
-            }
-        }
-        internal OutputAssemblyType outputType = OutputAssemblyType.Library;
-        internal bool outputTypeSpecified = false;
+        public OutputAssemblyType OutputType { get; set; } = OutputAssemblyType.Library;
 
         /// <summary>
         /// Flag to pass the resulting types along.
         /// </summary>
-        [Parameter()]
+        [Parameter(ParameterSetName = FromSourceParameterSetName)]
+        [Parameter(ParameterSetName = FromMemberParameterSetName)]
+        [Parameter(ParameterSetName = FromPathParameterSetName)]
+        [Parameter(ParameterSetName = FromLiteralPathParameterSetName)]
+        [Parameter(ParameterSetName = FromAssemblyNameParameterSetName)]
         public SwitchParameter PassThru { get; set; }
 
+        /// <summary>
+        /// Flag to compile without loading compiled assembly.
+        /// </summary>
+        [Parameter(ParameterSetName = FromSourceParameterSetName)]
+        [Parameter(ParameterSetName = FromMemberParameterSetName)]
+        [Parameter(ParameterSetName = FromPathParameterSetName)]
+        [Parameter(ParameterSetName = FromLiteralPathParameterSetName)]
+        public SwitchParameter CompileOnly { get; set; }
```

Comment:
What if `-CompileOnly` and `-PassThru` are both specified?
As mentioned in some other places, I don't think we need this parameter.

Reply:
Currently `-PassThru` will be ignored.
The switch came from https://github.com/PowerShell/PowerShell/issues/5725#issuecomment-354164119

Reply:
As I stated in comments above https://github.com/PowerShell/PowerShell/pull/6141#discussion_r173548138, when using `-OutputAssembly`, `Add-Type` only compiles the assembly without loading it. So I don't think `-CompileOnly` is really needed.

Reply:
Currently we always load [generated assembly](https://github.com/PowerShell/PowerShell/blob/c1c5344a8897262433141ecbc13bb06ac2c4bbef/src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs#L1098). So with `-CompileOnly ` we address new scenario.

Reply:
Fixed.